packages/browser-wallet/src/popup/page-layouts/MainLayout/IdentityList/IdentityList.tsx

```diff
@@ -54,20 +54,20 @@ type Props = ClassName & {
 const IdentityList = forwardRef<HTMLDivElement, Props>(({ className, onSelect }, ref) => {
     const identities = useAtomValue(identitiesAtom);
     const accountsPerIdentity = useAtomValue(accountsPerIdentityAtom);
-    const [selectedIdentityIndex, setSelectedIdentityId] = useAtom(selectedIdentityIndexAtom);
+    const [selectedIdentityIndex, setSelectedIdentityIndex] = useAtom(selectedIdentityIndexAtom);
     const nav = useNavigate();
     const { t } = useTranslation('mainLayout');
 
     return (
-        <EntityList<Identity>
+        <EntityList<Identity & { i: number }>
             className={className}
-            onSelect={(a) => {
-                setSelectedIdentityId(a.index);
+            onSelect={(id) => {
+                setSelectedIdentityIndex(id.i);
                 onSelect();
             }}
             onNew={() => nav(absoluteRoutes.home.identities.add.path)}
-            entities={identities}
-            getKey={(a) => a.index}
+            entities={identities.map((id, i) => ({ ...id, i }))}
+            getKey={(a) => `${a.index}-${a.provider}`}
```

Comment:
It doesn't affect the functionality, but it would be nicer to have the `providerIndex` first as we index `providerIndex/identityIndex`.

Reply:
:thinking: sure.